pyuvdata/uvdata/ms.py

```diff
@@ -126,11 +886,715 @@ def find_not_ascii(s):
+        # the weights and noise for all channels in each row in the MAIN table, which
+        # we will roughly calculate in temp_weights below.
+        temp_weights = np.zeros((self.Nblts * self.Nspws, self.Npols), dtype=float)
+
+        for attr, col in zip(attr_list, col_list):
+            val_dict = {}
+            temp_array = getattr(self, attr)
+
+            if not self.future_array_shapes:
+                temp_array = np.squeeze(temp_array, axis=1)
+
+            for idx, spw_id in enumerate(self.spw_array):
+                if self.flex_spw:
+                    sel_mask = self.flex_spw_id_array == spw_id
+                else:
+                    sel_mask = np.ones(self.Nfreqs, dtype=bool)
+
+                temp_dict = {
+                    "%016d"
+                    % ((self.Nspws * jdx) + 1 + idx): temp_array[None, jdx, sel_mask, :]
+                    for jdx in np.arange(self.Nblts)
+                }
+                if col == "WEIGHT_SPECTRUM":
+                    temp_weights[idx :: self.Nspws] = np.median(
+                        temp_array[:, sel_mask], axis=1
+                    )
+
+                val_dict.update(temp_dict)
+            ms.putvarcol(col, {key: val_dict[key] for key in sorted(val_dict.keys())})
+
+        # TODO: If/when UVData objects can store visibility noise estimates, update
+        # the code below to capture those.
+        ms.putcol("WEIGHT", temp_weights)
+        ms.putcol("SIGMA", np.power(temp_weights, -0.5, where=temp_weights != 0))
+
+        ms.putcol("ANTENNA1", np.repeat(self.ant_1_array, self.Nspws))
+        ms.putcol("ANTENNA2", np.repeat(self.ant_2_array, self.Nspws))
+        ms.putcol("INTERVAL", np.repeat(self.integration_time, self.Nspws))
+        ms.putcol("EXPOSURE", np.repeat(self.integration_time, self.Nspws))
+
+        ms.putcol("DATA_DESC_ID", np.tile(np.arange(self.Nspws), self.Nblts))
+
+        # Note that the default for MS is UTC, which is the same as UVData
+        ms.putcol(
+            "TIME",
+            np.repeat(
+                Time(self.time_array, format="jd").mjd * 3600.0 * 24.0, self.Nspws,
+            ),
+        )
+
+        # FITS uvw direction convention is opposite ours and Miriad's.
+        # CASA's convention is unclear: the docs contradict themselves,
+        # but after a question to the helpdesk we got a clear response that
+        # the convention is antenna2_pos - antenna1_pos, so the convention is the
+        # same as ours & Miriad's
+        ms.putcol("UVW", np.repeat(self.uvw_array, self.Nspws, axis=0))
+        if self.multi_phase_center:
+            # We have to do an extra bit of work here, because CASA won't accept arb
+            # values for field ID (rather, the ID number matches to the row number in
+            # the FIELD subtable). When we write out the fields, we use sort so that
+            # we get reproduce the same ordering here
```

Comment:
we get reproduce -> we reproduce

Reply:
English is hard. Commenting code -- harder. Fixed in 3068e067.